spec/rails_cursor_pagination/cursor_spec.rb

```diff
@@ -0,0 +1,203 @@
+# frozen_string_literal: true
+
+RSpec.describe RailsCursorPagination::Cursor do
+  describe '.encode' do
+    let(:record) { Post.create! id: 1, author: 'John', content: 'Post 1' }
+
+    context 'when ordering by id implicitly' do
+      subject(:encoded) do
+        described_class.encode(record: record)
+      end
```

Comment:
Thinking about it more, I actually believe given the alternatives, the easiest and cleanest way would be to remove the class-method `.encode` and just call `Cursor.new(...).encode` instead of the `Cursor.encode(...)` shortcut. 

I believe that all methods that are accessible from the outside should have proper test coverage – even more so in a gem that might be used (and abused or monkey-patched 🙈) by others. Then it's important to make sure that the methods we offer to the outside do exactly what we want them to and that this behavior stays consistent between versions (unless it's a conscious decision to change it, of course). But therefore, it also makes sense to keep the public API as small as possible.

And I would refrain from using something like `.send` meta-programming just to reduce the API of the class. That makes understanding the code especially for beginners (as well as some more naïve debuggers or IDEs) unnecessarily hard.

But I actually also see explicitly calling `Cursor.new(...).encode` as a benefit: it shows that there's an actual cursor object that's being created and used – something that might help others in future if they should want to implement their own cursor classes or use the Cursor from the outside.

So **tl;dr**: My suggestion would be to get rid of `Cursor.encode` in favor of just using (and testing) the instance method `Cursor#encode` 🙂 

What do you think??

Reply:
Looks great, thanks a lot for the changes!! 😊